### ISSUE TITLE
unflagging/flagging message in outlook is not transferred to roundcube

### ISSUE THREAD

**Issue by rcubetrac** [thomascube/roundcube-trac-import#121]
**Reported by backi on 2 Jul 2008 20:19 UTC**

If I flag/unflag a message in outlook, the changes were not applied in roundcube.

only if the message was already flagged by roundcube, i can unflag it in outlook and the changes were applied in roundcube.

i hope that wasnt too complicated :-)

dont know if this is a bug of outlook (2007) or of roundcube

Migrated-From: http://trac.roundcube.net/ticket/1485186

**Comment by rcubetrac** [comment]
**Comment by tensor on 20 Sep 2008 16:56 UTC**
Tested with Outlook 2007 SP1 (12.0.6316.5000) and RC svn.

Outlook 2007 sees both kinds of transitions from RC: flagged->unflagged and unflagged->flagged.

RC only sees the first transition of any kind or does not see at all.

If I disable cache, RC sees all transitions.

RC updates the state of the flag only when entering folder. If I click "Check for new messages", the flag state is not updated even with caching disabled.

**Comment by rcubetrac** [comment]
**Comment by thomascube on 1 May 2009 16:13 UTC**
We need more investigation how Outlook stores flags via IMAP. It looks to me like Outlook reads and displays "foreign" flags but does not set them the same way. Could anybody post some IMAP conversation when flagging/unflagging a message with Outlook?

**Comment by rcubetrac** [comment]
**Comment by sebs on 25 Oct 2009 11:27 UTC**
outlook 2007:[6767 UID STORE 9534 FLAGS (\Seen \Flagged)
IMAP    Response: * 240 FETCH (UID 9534 FLAGS (\Flagged \Seen))
```

unflag:[[br]([br]]
[[br]]
flag:[[br]]
```
IMAP	Request:)]
```
IMAP	Request: a7st UID STORE 9534 FLAGS (\Seen)
IMAP    Response: * 240 FETCH (UID 9534 FLAGS (\Seen))
```

**Comment by rcubetrac** [comment]
**Comment by alec on 6 May 2010 16:38 UTC**
Flags are set in standard way. As tensor write, flags are not updated only when using "check for new messages". This feature is checking only for existance of new messages not for state (flag/status) changes.

**Comment by rcubetrac** [comment]
**Comment by thomascube on 16 Nov 2011 17:12 UTC**
Maybe a caching issue on the Roundcube side? We meanwhile re-implemented the entire caching with better handling for flags. Please recheck with the latest version from SVN.

**Comment by rcubetrac** [comment]
**Comment by alec on 16 Nov 2011 17:17 UTC**
No, this is still not resolved. \Flagged flag changes are not a subject for check_recent action still.

**Comment by rcubetrac** [comment]
**Comment by vanmeeuwen on 27 Jun 2012 08:19 UTC**
Indeed, this is not resolved yet, for at least \Seen flags that are updated from another client.

The messages unread counter (in the folder list) does appropriately display the new unread count, but the message list still displays individual messages as if they were unread (i.e., in bold).

**Comment by rcubetrac** [comment]
**Comment by thomascube on 5 Aug 2012 11:09 UTC**
Reschedule

**Comment by rcubetrac** [comment]
**Comment by alec on 22 Mar 2013 12:09 UTC**
It is not a simple task, but I'm going to finally work on this for 1.0.

**Comment by rcubetrac** [comment]
**Modified by alec on 1 Dec 2013 10:02 UTC**

**Comment by rcubetrac** [comment]
**Comment by alec on 23 Dec 2013 10:34 UTC**
5 year, ah? Fixed in ac0fc383fd43e8955b0ab22f70463159b14c74b0.
